thoth/common/openshift.py

```diff
@@ -1608,6 +1609,43 @@ def get_workflow(
 
         return wf
 
+    def get_workflow_status_report(
+        self,
+        workflow_id: Optional[str] = None,
+        label_selector: Optional[str] = None,
+        namespace: Optional[str] = None,
+    ) -> Tuple[Dict[str, Any], int]:
+        """Get workflow status report, derived from workflow status."""
+        if workflow_id is None:
+            return {"error": "No analysis id provided"}, 400
+        try:
+            wf_status = self.get_workflow_status(
+                name=workflow_id, label_selector=label_selector, namespace=namespace
+            )
+            status = {}
+            # the Job has not started (yet) but the Workflow has been submitted
+            # if the Workflow fails, the status will contain the finished
+            # time of the workflow
+            status["status"], code = (
+                {
+                    "finished_at": wf_status.get("finishedAt"),
+                    "reason": None,
+                    "started_at": wf_status.get("startedAt"),
+                    "state": wf_status.get("phase", "Pending"),
+                },
+                200,
+            )
+        except NotFoundException:
+            return (
+                {
+                    "error": f"Requested status for analysis {workflow_id!r} was not found"
+                },
+                404,
```

Comment:
I think `400` (bad request) is more apt.

Reply:
400 is used when the request is a malformed type like passing an int instead, but 404 seems the right choice considering the format is correct, but resource was not found.

Reply:
Please move HTTP status codes out of this module. They can be added on API services (user-api/management-api/...), this module should not capture them. For example, these functions can be used also on metrics exporter to export info about running/pending/... workflows and HTTP status code is not relevant there.